lib/unhangout-routes.js

```diff
@@ -788,6 +789,121 @@ module.exports = {
         // Starting and stopping events
         //
 
+        function generateUserData(attrs) {   
+
+            var event = db.events.get(attrs.id);
+            var history = event.get("history");
+
+            var userData = [];
+
+            _.each(history.event, function(elapsedTime, userId) {
+                
+                //Get the user object for a specific ID 
+                var user = db.users.get(userId);
+
+                // Find all userIDs for users who shared sessions with me at this event.
+                var userIds = event.getUserIdsSharingSessionsWith(userId);
+                
+                // get user objects for the IDs.
+                var users = _.map(userIds, function(userId) { 
+                    return db.users.get(userId); 
+                });
+
+                userData.push({ user: user, users: users });
+
+            });
+
+            return userData; 
+
+        }
+
+        app.get("/followup/event/:id/participant_:participantIndex", utils.ensureAuthenticated, utils.ensureSuperuser, function(req, res) {
+
+            //Call generate user data function to grab history of event participants
+            var userData = generateUserData(req.params);   
+            userData.unshift(null);
+
+            if(userData) {
+                res.render("email_preview_template.ejs", {
+                    participantIndex: req.params.participantIndex,
+                    userData: userData,
+                    userDataCount: userData.length - 1,
+                    event: db.events.get(req.params.id)
+                });
+            }
+        });
+
+        app.post("/followup/event/:id/sent/", utils.ensureAuthenticated, utils.ensureSuperuser, function(req, res) {
+
+            //Call generate user data function to grab history of event participants
+            var userData = generateUserData(req.params);
+
+            var mandrill_client = new mandrill.Mandrill('YOUR MANDRILL API KEY');
+
+            var message = {}; 
+            var emailsSent = 0;
+
+            //Execute a loop to send emails
+            _.each(userData, function(user, participantIndex) {
+                
+                res.render("email_template.ejs", {
+                    event: db.events.get(req.params.id),
+                    userData: userData,
+                    participantIndex: participantIndex
+
+                }, function(err, html) {
+
+                    if (err) {
+                        logger.error("Error rendering email body", err);
+                        return res.send(500, "Server error");
```

Comment:
Email sending loop now uses caolan's async module. Referring to the documentation indicated in this link https://github.com/caolan/async#each And corresponding changes are indicated in this commit
https://github.com/drewww/unhangout/commit/40b83f54cd95a8588c4ed4098c66134886b760a1